src/utils/application.data.ts

```diff
@@ -42,3 +43,28 @@ export const mapFieldsToDataObject = (data: any, htmlFields: string[], dataModel
 export const mapDataObjectToFields = (data: any, htmlFields: string[], dataModelKeys: string[]) => {
   return htmlFields.reduce((o, key, i) => Object.assign(o, { [key]: data[dataModelKeys[i]] }), {});
 };
+
+export const removeFromApplicationData = (session: Session | undefined, key: string, id: string | undefined) => {
+  const appData: ApplicationData = getApplicationData(session);
+
+  if (id && appData && appData[key]) {
+    const index = appData[key].findIndex( object => object[ID] === id );
+
+    if (index !== -1) {
+      appData[key].splice(index, 1);
+      setExtraData(session, appData);
+    }
```

Comment:
Should we maybe log it if a data item isn't found with the id? It should never happen but if it did it indicates something's gone wrong.

Reply:
good idea, i'll log and throw error here if we are trying to remove something that cannot be found